lib/block_ccm.mli

```diff
@@ -1,9 +1,12 @@
-module Make (B: Mirage_types.BLOCK
-             with type 'a io = 'a Lwt.t
-              and type page_aligned_buffer = Cstruct.t) : sig
-  include Mirage_types.BLOCK
-    with type 'a io = 'a Lwt.t
-     and type page_aligned_buffer = Cstruct.t
-
-  val connect : ?maclen:int -> ?nonce_len:int -> key:Cstruct.t -> B.t -> t io
+module Make (B: Mirage_block_lwt.S) : sig
+
+  type error = private [> Mirage_device.error | `DecryptError ]
```

Comment:
If you want the clients to known exactly all the possible errors, you can just write:
```
type error = [ Mirage_device.error | `DecryptError ]
```